### PR TITLE
Create the canary kops version of `pull-kubernetes-e2e-gce` job for k/k

### PR DESCRIPTION
**Pull request by upodroid**
/cc @dims @BenTheElder @aojea @hakman 

This PR creates the core gce e2e test we run in k/k presubmits. I also created the serial test as well to test fixes on PRs.

1. pull-kubernetes-e2e-gce-cos-kubetest2 job is no longer required, `kubetest2 gce` will be replaced by `kubetest2 kops` jobs
2. `pull-kubernetes-e2e-kops-aws` has no build history. https://prow.k8s.io/job-history/gs/kubernetes-jenkins/logs/pull-kubernetes-e2e-kops-aws

Part of https://github.com/kubernetes/enhancements/issues/4224

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml [hunk 1]
```diff
@@ -300,46 +252,40 @@ presubmits:
...
-            - --gcp-zone=us-west1-b
-            - --ginkgo-parallel=30
-            - --provider=gce
-            - --stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce-canary
-            - --test_args=--ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
-            - --timeout=80m # thinking about making this longer? don't! 80m is a hard cap, and should get down to no more than 60m.
+            - bash
+            - -c
+            - |
+              ARGS="--set=spec.containerd.runc.version=1.1.9 --set=spec.containerd.version=1.7.5"
```
Comment (hakman) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:275]: runc v1.1.9 support needs to be added. See https://github.com/kubernetes/kops/pull/16043.
Maybe also switch to containerd v1.7.7.

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml [hunk 2]
```diff
@@ -300,46 +252,40 @@ presubmits:
...
     decoration_config:
-      timeout: 105m
+      timeout: 110m
     path_alias: k8s.io/kubernetes
     extra_refs:
     - org: kubernetes
-      repo: release
+      repo: kops
       base_ref: master
-      path_alias: k8s.io/release
```
Comment (aojea) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:315]: why did this mount release? is ok to swap release by kops?

Comment (upodroid) [reply config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:315]: Yes, kops implementes krel to build and stage kube builds

https://github.com/kubernetes/kops/blob/master/tests/e2e/kubetest2-kops/builder/build.go#L67

**Comment by aojea** [comment]
Lgtm, has conflicts though

**Comment by hakman** [comment]
@ameukam Any idea how to remove or re-trigger the `needs-rebase` check?

**Pull request merged**